### PR TITLE
feat: add implementation for use case requests

### PR DESCRIPTION
**Pull request by lucas-soranzo**

### CODE REVIEW

File: src/bulkUseCaseRequest.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      payload = JSON.parse(req.body);
+    }
+    const trialRequest = new TestBoxBulkUseCaseRequest(payload);
+    await trialRequest.verifyToken(
+      req.headers.authorization.replace("Bearer ", "")
+    );
+    return trialRequest;
+  }
+
+  async verifyToken(authToken: string): Promise<boolean> {
```
Comment (jakemwood) [inline src/bulkUseCaseRequest.ts]: 🧠 Might want to consider refactoring the `verifyToken` and `throwIfAuthNotValidated` into a new base class, `TestBoxRequest` or something, and then have `TestBoxBulkUseCaseRequest` and `TestBoxTrialRequest` inherit from it.

File: src/auth.ts [hunk 2]
```diff
@@ -1,35 +1,18 @@
...
-  if (!publicKey) {
-    return false;
-  }
   try {
-    const importedKey = await jose.importSPKI(publicKey, "RS256");
-    const results = await jose.jwtVerify(token, importedKey, {
+    const publicKey = jose.createRemoteJWKSet(
+      new URL(
+        process.env.TBX_JWKS_URL ||
+          "https://partners-well-known.s3.us-west-2.amazonaws.com/.well-known/jwks.json"
```
Comment (n1tsua) [inline src/auth.ts]: I think we should put testbox somewhere in this url, or we should alias it at trial.testbox.com/.well-known/jwks.json

**Comment by kiimmmmyy** [comment]
@lucas-soranzo ???

**Comment by lucas-soranzo** [comment]
> @lucas-soranzo ???

This was part of outloading things to be deal with by the partner. So in this case partner would be able to do saturn's work on their end.

I'm closing this as its 2022 and we have dropped the idea

**Pull request closed without merging**